pymc_marketing/mmm/lift_test.py

```diff
@@ -0,0 +1,366 @@
+    name: str = "lift_measurements",
+) -> None:
+    """Add lift measurements to the likelihood of the model.
+
+    Args:
+        df_lift_test: DataFrame with lift test results.
+        variable_mapping: Dictionary of variable names to dimensions.
+        saturation_function: Function that takes spend and returns saturation.
+        model: PyMC model with arbitrary number of coordinates.
+        dist: PyMC distribution to use for the likelihood.
+        name: Name of the likelihood.
+
+    Returns:
+        None
+
+    """
+    required_columns = ["x", "delta_x", "delta_y", "sigma"]
+
+    missing_cols = set(required_columns).difference(df_lift_test.columns)
+    if missing_cols:
+        raise KeyError(f"Missing from DataFrame: {list(missing_cols)}")
+
+    model = pm.modelcontext(model)
+
+    var_names = list(variable_mapping.values())
+    indices = indices_from_lift_tests(df_lift_test, model, var_names)
+
+    x_before = df_lift_test["x"].to_numpy()
+    x_after = x_before + df_lift_test["delta_x"].to_numpy()
+
+    kwargs = {
+        name: index_variable(
+            var_dims=model.named_vars_to_dims[var_name],
+            var=model[var_name],
+            indices=indices,
+        )
+        for name, var_name in variable_mapping.items()
+    }
+
+    partial_saturation_function = partial(saturation_function, **kwargs)
+    empirical_lift = empirical_lift_measurements(
+        x_before, x_after, partial_saturation_function
+    )
+
+    dist(
+        name=name,
+        mu=empirical_lift,
+        sigma=df_lift_test["sigma"].to_numpy(),
+        observed=df_lift_test["delta_y"].to_numpy(),
+    )
+
+
+def add_menten_empirical_lift_measurements_to_likelihood(
+    df_lift_test: pd.DataFrame,
+    alpha_name: str,
+    lam_name: str,
+    dist=pm.Gamma,
+    model: Optional[pm.Model] = None,
+    name: str = "lift_measurements",
+) -> None:
+    """Add empirical lift measurements to the likelihood of the model.
+
+    Args:
+        df_lift_test: DataFrame with lift test results.
+        alpha_name: Name of the alpha parameter in the model.
+        lam_name: Name of the lambda parameter in the model.
+        dist: PyMC distribution to use for the likelihood.
+        model: PyMC model with date and channel coordinates.
+        name: Name of the likelihood.
+
+    Returns:
+        None
+
+    """
+    variable_mapping = {
+        "alpha": alpha_name,
+        "lam": lam_name,
+    }
+
+    add_lift_measurements_to_likelihood(
+        df_lift_test,
+        variable_mapping,
+        saturation_function=michaelis_menten,
+        model=model,
+        dist=dist,
+        name=name,
+    )
+
+
+def add_logistic_empirical_lift_measurements_to_likelihood(
+    df_lift_test: pd.DataFrame,
+    lam_name: str,
+    beta_name: str,
+    dist=pm.Gamma,
+    model: Optional[pm.Model] = None,
+    name: str = "lift_measurements",
+) -> None:
+    """Add empirical lift measurements to the likelihood of the model.
+
+    Args:
```

Comment:
I think we are following the numpy docstring format right? https://numpydoc.readthedocs.io/en/latest/format.html